temporalio/client.py

```diff
@@ -2758,6 +2758,12 @@ class WorkflowExecution:
     parent_run_id: Optional[str]
     """Run ID for the parent workflow if this was started as a child."""
 
+    root_id: Optional[str]
+    """ID for the root workflow"""
+
+    root_run_id: Optional[str]
+    """Run ID for the root workflow"""
```

Comment:
So _technically_ this is an incompatible change since users can technically instantiate this object themselves which some users may do in a mock client situation (I know they do in other languages). Two options:

1. Add a note to the docstring of this class (and `WorkflowExecutionDescription`) that the constructor may change in backwards incompatible ways
2. Add defaults for these values

I'm on the fence on which I prefer. @dandavison or @THardy98 - preference?

Reply:
I think I'd lean toward the latter while we can and introduce the former when we have to. I suppose default values for these fields would be `None`?

Reply:
Yup

Reply:
Added defaults

Reply:
Removed defaults - linter complains about having fields with default values before fields without defaults, and this class is inherited by `WorkflowExecutionDescription` below, making it annoying to fix.

Reply:
Yes but it still conflicts with the non-defaults fields in `WorkflowExecutionDescription`, unless I'm misunderstanding you